rest-api/src/main/java/com/cloudbees/workflow/flownode/FlowNodeUtil.java

```diff
@@ -249,20 +379,31 @@ public static FlowNode getStageEndNode(FlowNode stageStartNode) {
         return allNodesSorted.get(allNodesSorted.size() - 1);
     }
 
+    @CheckForNull
     /**
      * Get the last node to start in a flow.
      * @param execution The flow execution.
      * @return The last node to start in the flow.
      */
     public static FlowNode getFlowEndNode(FlowExecution execution) {
+        if (execution == null) {
+            return null;
+        }
+
         // If there's no next stage, then use the last node in the workflow.
         List<FlowNode> endNodes = new ArrayList<FlowNode>(execution.getCurrentHeads());
         sortNodesById(endNodes);
 
+        // We do not actually need to sort this, we can just the get the head of the execution
```

Comment:
Agreed, changed in latest commit.   This will all go away with #6 though, almost all the stage analysis is restructured completely and most of the methods become needless.